Mlem/App/Views/Root/Tabs/Feeds/PostGridView+Logic.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  PostGridView+Logic.swift
+//  Mlem
+//
+//  Created by Eric Andrews on 2024-08-12.
+//
+
+import Foundation
+
+extension PostGridView {
+    var standardActions: [any Action] {
+        var ret: [any Action] = .init()
+        
+        ret.append(BasicAction(
+            id: "read",
+            isOn: showRead,
+            label: showRead ? "Hide Read" : "Show Read",
+            color: palette.primary,
+            icon: Icons.read
+        ) {
+            showRead = !showRead
+        })
+        
+        ret.append(ActionGroup(
+            isOn: true,
+            label: "Post Size",
+            color: palette.primary,
+            isDestructive: false,
+            icon: Icons.postSizeSetting,
+            displayMode: .disclosure, children: postSizeActions
+        ))
+        
+        return ret
+    }
+    
+    var postSizeActions: [BasicAction] {
+        var ret: [BasicAction] = .init()
+        
+        PostSize.allCases.forEach { size in
+            ret.append(.init(
+                id: size.rawValue,
+                isOn: postSize == size,
+                label: size.label,
+                color: palette.primary,
+                isDestructive: false,
+                icon: size.icon(filled: postSize == size),
+                enabled: true
+            ) {
+                postSize = size
+            })
+        }
+        
+        return ret
+    }
+}
```

Comment:
I'm not sure we should be using actions when we only need them in one context. This is a lot of extra code that could be avoided by using a simple SwiftUI picker in the toolbar:

```swift
Picker("Post Size", systemImage: Icons.postSize, selection: $postSize) {
    ForEach(PostSize.allCases, id: \.self) { item in
        Label(item.label, systemImage: item.systemImage)
    }
}
```

Another advantage of using a Picker over multiple buttons is that it's got better accessibility support. With multiple buttons, there's no way for VO to know whether a button is "on" or not. I also think it's a bit clearer than using a symbol variant to indicate selection. What do you think?

(The same applies for the Hide/Show read button)

Reply:
Sure, that works for me